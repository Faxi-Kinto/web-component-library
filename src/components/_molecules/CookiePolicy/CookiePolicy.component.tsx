import React, { useState } from 'react';
import * as Styled from './CookiePolicy.styles';
import Text from '../../_atoms/Text';
import Button from '../../_atoms/Button';
import classNames from 'classnames';

export type CookieStatus = {
  acceptedCookies: boolean;
};

export type CookiePolicyProps = {
  cookieClassName?: string;
  title?: string;
  text?: JSX.Element;
  backgroundColor?: string;
  textColor?: string;
  buttonBackground?: string;
  buttonColor?: string;
  textClassName?: string;
  cookieTitleClassName?: string;
  acceptedCookies?: boolean;
  acceptButtonContent?: React.ReactNode;
  declineButtonContent?: React.ReactNode;
  userAction?: (cookieStatus: CookieStatus) => {};
};

const CookiePolicy: React.FC<CookiePolicyProps> = (
  props: CookiePolicyProps
): JSX.Element => {
  const {
    title,
    text,
    acceptButtonContent,
    declineButtonContent,
    backgroundColor = '#00708d',
    textColor = '#fff',
    buttonBackground = '#fff',
    buttonColor = '#00708d',
    textClassName,
    userAction,
    cookieClassName,
    cookieTitleClassName,
  } = props;

  const [isShowing, setIsShowing] = useState(true);

  const deleteCookies = () => {
    const cookies = document.cookie.split(';');
    for (let i = 0; i < cookies.length; i++) {
      const cook = cookies[i].split('=');
      document.cookie = cook[0] + '=;expires=Thu, 21 Sep 1979 00:00:01 UTC;';
    }
  };

  const declineCookies = () => {
    userAction && userAction({ acceptedCookies: false });
    if (isShowing) {
      setIsShowing(!isShowing);
      localStorage.setItem('showBanner', 'declined');
      deleteCookies();
    }
  };

  const acceptCookies = () => {
    userAction && userAction({ acceptedCookies: true });
    if (isShowing) {
      setIsShowing(!isShowing);
      localStorage.setItem('showBanner', 'accepted');
    }
  };

  return (
    <Styled.Container
      backgroundColor={backgroundColor}
      textColor={textColor}
      buttonBackground={buttonBackground}
      buttonColor={buttonColor}
    >
      {isShowing ? (
        <div className={classNames(['cookie-policy', cookieClassName])}>
          <Text.Heading
            level="3"
            className={classNames([
              'cookie-policy__title',
              cookieTitleClassName,
            ])}
          >
            {title}
          </Text.Heading>
          <Text.Body className="cookie-policy__text">
            {text &&
              React.cloneElement(text, {
                className: classNames([
                  'cookie-policy__text__body',
                  textClassName,
                ]),
              })}
          </Text.Body>
          <Button
            className="cookie-policy__accept-button"
            onClick={acceptCookies}
          >
            {acceptButtonContent}
          </Button>
          <Button
            className="cookie-policy__decline-button"
            onClick={declineCookies}
          >
            {declineButtonContent}
          </Button>
        </div>
      ) : null}
    </Styled.Container>
  );
};

export default CookiePolicy;
