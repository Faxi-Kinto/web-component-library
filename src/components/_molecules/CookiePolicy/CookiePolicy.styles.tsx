import styled from 'styled-components';
import { pxToRem, padding } from '@faxi/web-css-utilities';

type CookiePolicyStylingProps = {
  bgColor?: string;
  textColor?: string;
  buttonColor?: string;
  buttonBackground?: string;
};

export const Container = styled.div`
  .cookie-policy {
    ${padding(pxToRem('20px'), pxToRem('50px'))};
    background-color: ${(props: CookiePolicyStylingProps) =>
      `${props.bgColor}`};

    &__title,
    &__text {
      display: block;
      margin-bottom: ${pxToRem('20px')};
      color: ${(props: CookiePolicyStylingProps) => `${props.textColor}`};
    }

    &__accept-button,
    &__decline-button {
      ${padding(pxToRem('10px'), pxToRem('35px'))};
      margin-right: ${pxToRem('20px')};
      background-color: ${(props: CookiePolicyStylingProps) =>
        `${props.buttonBackground}`};
      color: ${(props: CookiePolicyStylingProps) => `${props.buttonColor}`};
    }
  }
`;
