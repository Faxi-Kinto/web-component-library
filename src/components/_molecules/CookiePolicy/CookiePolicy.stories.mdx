import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import Button from '../../_atoms/Button';
import Icon from '../../_atoms/Icon';
import Text from '../../_atoms/Text';
import { action } from '@storybook/addon-actions';
import CookiePolicy from './CookiePolicy.component.tsx';

<Meta title="Core|Molecules/CookiePolicy" />

## CookiePolicy Component

#### Default

<Preview>
  <Story name="default">
    <CookiePolicy
      title={'Cookie Policy'}
      text={<Text.Body>We use cookies to ensure that we give you the best experience on our website. <Text.Link href="#" target="_blank">Read our privacy policy</Text.Link> and let us know that you agree to cookies.</Text.Body>}
      acceptButtonContent={<p>Accept</p>}
      declineButtonContent={<p>Reject</p>}
      userAction={action('cookie-click')}
    />
  </Story>
</Preview>

# Props

<Props of={CookiePolicy} />
