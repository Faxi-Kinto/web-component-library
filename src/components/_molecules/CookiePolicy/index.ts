import CookiePolicy from './CookiePolicy.component';

export default CookiePolicy;
