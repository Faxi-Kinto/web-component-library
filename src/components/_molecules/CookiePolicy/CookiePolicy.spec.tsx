import React from 'react';
import renderer from 'react-test-renderer';
import CookiePolicy from '.';

test('CookiePolicy renders correctly', () => {
  const component = renderer.create(
    <CookiePolicy
      title={'Cookie Policy'}
      text={`Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s.`}
      buttonContentF={<p>Accept</p>}
      buttonContentS={<p>Reject</p>}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
