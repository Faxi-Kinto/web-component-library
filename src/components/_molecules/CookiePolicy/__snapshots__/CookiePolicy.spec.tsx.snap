// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookiePolicy renders correctly 1`] = `
.c2 {
  padding: 0;
  margin: 0;
  font-size: 1rem;
  color: black;
  font-weight: 300;
  white-space: pre-wrap;
}

.c1 {
  padding: 0;
  margin: 0;
  font-size: 1.5rem;
  line-height: 2.125rem;
  font-weight: 500;
  color: black;
}

.c3 {
  padding: 0;
  margin: 0;
}

.c0 .cookie-policy {
  padding: 1.25rem 3.125rem 1.25rem 3.125rem;
  background-color: #00708d;
}

.c0 .cookie-policy__title,
.c0 .cookie-policy__text {
  display: block;
  margin-bottom: 1.25rem;
  color: #fff;
}

.c0 .cookie-policy__title > *,
.c0 .cookie-policy__text > *,
.c0 .cookie-policy__title a,
.c0 .cookie-policy__text a {
  color: #fff;
}

.c0 .cookie-policy__accept-button,
.c0 .cookie-policy__decline-button {
  padding: 0.625rem 2.1875rem 0.625rem 2.1875rem;
  margin-right: 1.25rem;
  background-color: #fff;
  color: #00708d;
}

.c4 {
  border: none;
  cursor: pointer;
  font: inherit;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  background: white;
  color: black;
}

.c4 p {
  margin: 0;
}

.c4:focus {
  outline: none;
}

.c4.button--secondary {
  background: transparent;
  font-size: 0.875rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  padding: 0;
}

.c4.button--secondary > * {
  margin: 0 0.75rem 0 0;
}

.c4.button--secondary > :last-child {
  margin: 0;
}

.c4.button--secondary p {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div
  className="c0"
>
  <div
    className="cookie-policy"
  >
    <h3
      className="c1 cookie-policy__title"
      color="black"
    />
    <span
      className="c2 cookie-policy__text"
      color="black"
      fontSize="16px"
    >
      <span
        className="c2 cookie-policy__text__body"
        color="black"
        fontSize="16px"
      >
        We use cookies to ensure that we give you the best experience on our website.
         
        <a
          className="c3"
          href="#"
          target="_blank"
        >
          Read our privacy policy
        </a>
         
        and let us know that you agree to cookies.
      </span>
    </span>
    <button
      className="c4 button cookie-policy__accept-button"
      height="fit-content"
      onClick={[Function]}
      type="button"
      width="fit-content"
    >
      <p>
        Accept
      </p>
    </button>
    <button
      className="c4 button cookie-policy__decline-button"
      height="fit-content"
      onClick={[Function]}
      type="button"
      width="fit-content"
    >
      <p>
        Reject
      </p>
    </button>
  </div>
</div>
`;
